File: cdk-from-cfn-testing/expected/vpc/golang/VpcConstruct.go
package main

import (
	cdk "github.com/aws/aws-cdk-go/awscdk/v2"
	ec2 "github.com/aws/aws-cdk-go/awscdk/v2/awsec2"
	"github.com/aws/constructs-go/constructs/v10"
	"github.com/aws/jsii-runtime-go"
)

type VpcConstructProps struct {
}

type VpcConstruct struct {
	constructs.Construct
}

func NewVpcConstruct(scope constructs.Construct, id string, props *VpcConstructProps) *VpcConstruct {
	construct := constructs.NewConstruct(scope, &id)

	vpc := ec2.NewCfnVPC(
		construct,
		jsii.String("VPC"),
		&ec2.CfnVPCProps{
			CidrBlock: jsii.String("10.42.0.0/16"),
			EnableDnsSupport: jsii.Bool(true),
			EnableDnsHostnames: jsii.Bool(true),
			Tags: &[]*cdk.CfnTag{
				&cdk.CfnTag{
					Key: jsii.String("cost-center"),
					Value: jsii.String("1337"),
				},
			},
		},
	)

	ec2.NewCfnSubnet(
		construct,
		jsii.String("Subnet1"),
		&ec2.CfnSubnetProps{
			AvailabilityZone: cdk.Fn_Select(jsii.Number(0), cdk.Fn_GetAzs(jsii.String(""))),
			CidrBlock: cdk.Fn_Select(jsii.Number(0), cdk.Fn_Cidr(vpc.AttrCidrBlock(), jsii.Number(6), jsii.String("8"))),
			VpcId: vpc.Ref(),
		},
	)

	ec2.NewCfnSubnet(
		construct,
		jsii.String("Subnet2"),
		&ec2.CfnSubnetProps{
			AvailabilityZone: cdk.Fn_Select(jsii.Number(1), cdk.Fn_GetAzs(jsii.String(""))),
			CidrBlock: cdk.Fn_Select(jsii.Number(1), cdk.Fn_Cidr(vpc.AttrCidrBlock(), jsii.Number(6), jsii.String("8"))),
			VpcId: vpc.Ref(),
		},
	)

	ec2.NewCfnSubnet(
		construct,
		jsii.String("Subnet3"),
		&ec2.CfnSubnetProps{
			AvailabilityZone: cdk.Fn_Select(jsii.Number(2), cdk.Fn_GetAzs(jsii.String(""))),
			CidrBlock: cdk.Fn_Select(jsii.Number(2), cdk.Fn_Cidr(vpc.AttrCidrBlock(), jsii.Number(6), jsii.String("8"))),
			VpcId: vpc.Ref(),
		},
	)

	return &VpcConstruct{
		Construct: construct,
	}
}

File: cdk-from-cfn-testing/expected/vpc/csharp/VpcConstruct.cs
using Amazon.CDK;
using Amazon.CDK.AWS.EC2;
using Constructs;
using System.Collections.Generic;

namespace VpcConstruct
{
    public class VpcConstructProps
    {
    }

    public class VpcConstruct : Construct
    {
        public VpcConstruct(Construct scope, string id, VpcConstructProps props = null) : base(scope, id)
        {

            // Resources
            var vpc = new CfnVPC(this, "VPC", new CfnVPCProps
            {
                CidrBlock = "10.42.0.0/16",
                EnableDnsSupport = true,
                EnableDnsHostnames = true,
                Tags = new []
                {
                    new CfnTag
                    {
                        Key = "cost-center",
                        Value = "1337",
                    },
                },
            });
            var subnet1 = new CfnSubnet(this, "Subnet1", new CfnSubnetProps
            {
                AvailabilityZone = Fn.Select(0, Fn.GetAzs("")),
                CidrBlock = Fn.Select(0, Fn.Cidr(vpc.AttrCidrBlock, 6, "8")),
                VpcId = vpc.Ref,
            });
            var subnet2 = new CfnSubnet(this, "Subnet2", new CfnSubnetProps
            {
                AvailabilityZone = Fn.Select(1, Fn.GetAzs("")),
                CidrBlock = Fn.Select(1, Fn.Cidr(vpc.AttrCidrBlock, 6, "8")),
                VpcId = vpc.Ref,
            });
            var subnet3 = new CfnSubnet(this, "Subnet3", new CfnSubnetProps
            {
                AvailabilityZone = Fn.Select(2, Fn.GetAzs("")),
                CidrBlock = Fn.Select(2, Fn.Cidr(vpc.AttrCidrBlock, 6, "8")),
                VpcId = vpc.Ref,
            });
        }
    }
}
